evennia/contrib/currency.py

```diff
@@ -0,0 +1,374 @@
+
+    def __len__(self):
+        # Return the number of values in 'attr_dict'.
+        return len(self.attr_dict)
+
+    def __setattr__(self, key, value):
+        # Return an error message if attrib is assigned directly
+        if key in ('attr_dict', 'cache'):
+            super(CurrencyHandler, self).__setattr__(key, value)
+        else:
+            raise CurrencyException(
+                "Currency object not settable. Assign one of " + 
+                "the monetary conversion properties instead."
+            )
+
+    def __setitem__(self, key, value):
+        # Returns error message if currency objects are assigned directly.
+        return self.__setattr__(key, value)
+
+    def __getattr__(self, currency):
+        # Returns currency instances accessed as attributes.
+        return self.get(currency)
+
+    def __getitem__(self, currency):
+        # Returns 'Currency' instances accessed as dict keys.
+        return self.get(currency)
+
+    def get(self, currency):
+        """
+        Args: 
+            currency (str): key from the currency dict containing config data
+            for the currency type. "all" returns a list of all currency keys
+        Returns:
+            ('Currency' or 'None'): named Currency class or None if currency key
+            is not found in currency collection.
+        """
+        if currency not in self.cache:
+            if currency not in self.attr_dict:
+                return None
+            data = self.attr_dict[currency]
+            self.cache[currency] = Currency(data)
+        return self.cache[currency]
+
+    def add(self, key, name, value, amt):
+        # Create a new Currency and add it to the handler.
+        if key in self.attr_dict:
+            raise CurrencyException("Currency '{}' already exists.".format(key))
+
+        currency = dict(name=name,
+                     value=value, 
+                     amt=amt)
+
+        self.attr_dict[key] = currency
+
+    def remove(self, currency):
+        # Remove a Currency type from the handler's parent object.
+        if currency not in self.attr_dict:
+            raise CurrencyException("Currency not found: {}".format(currency))
+
+        if currency in self.cache:
+            del self.cache[currency]
+        del self.attr_dict[currency]
+
+    def clear(self):
+        # Remove all Currency from the handler's parent object.
+        for currency in self.all:
+            self.remove(currency)
+
+    @property
+    def all(self):
+        # Return a list of all currency in this CurrencyHandler.
+        return self.attr_dict.keys()
+
+    @property
+    def contents(self):
+        # Return a formatted list of currency and amounts in this CurrencyHandler.
+        liststr = ""
+        for key in self.attr_dict.keys():
+            cur = Currency(self.attr_dict[key])
+            if cur.amt > 0:
+                liststr += "{}, ".format(cur)
+        if len(liststr.strip()) > 0: return liststr[:-2]

```

Comment:
According to PEP8 (see the style guide), these if...else  statements should not be written on one line but split across two lines for readability.